src/utils.rs

```diff
@@ -155,3 +156,34 @@ pub fn timed_read<T, F: FnOnce() -> T>(label: &str, f: F, total: &mut Duration)
     crate::utils::log_mem(format!("{label} loaded in {elapsed:?}"));
     res
 }
+
+/// Calculate modular inverse using the extended Euclidean algorithm
+/// Exposed for reuse by other CRT components.
+pub fn mod_inverse(a: &BigUint, m: &BigUint) -> Option<BigUint> {
```

Comment:
why this have to be in utils?

Reply:
This is because this function is used both in the crt/montgomery.rs and crt/mod.rs.